### PR TITLE
[hotfix] disable make public registrations private

### PR DESCRIPTION
**Pull request by HarryRybacki**
## Purpose
Registrations now have the ability to obtain DOIs and ARK IDs. With this and upcoming behavior changes with Retractions and Embargoes, public registrations should not be able to be made private unless they were created before a predetermined cutoff date.

## Changes
Registrations created before the cutoff date will have no behavioral changes.

Registrations created after the cutoff date will no longer be able to be made private **after** they have been made public. That cutoff date, **REGISTRATION_CUTOFF_DATE**, is set in `defaults.py`

## Side Effects
None

## Notes
Closes-issue: #2670

### CODE REVIEW

**Comment by sloria** [comment]
Can you add comments in the places that should be removed once we disable 'privatizing' public registrations for all registrations?

File: website/project/views/node.py [hunk 1]
```diff
@@ -475,6 +475,9 @@ def project_set_privacy(auth, node, **kwargs):
     if permissions is None:
         raise HTTPError(http.BAD_REQUEST)
 
+    if permissions == 'private' and node.is_registration and not node.registered_before_cutoff_date:
+        raise HTTPError(http.BAD_REQUEST)
```
Comment (sloria) [inline website/project/views/node.py]: Please add a useful message by passing `data=dict(message_long='Public registrations cannot be made private.')`

**Comment by sloria** [comment]
Pass finished

**Comment by HarryRybacki** [comment]
@sloria comments added. Jeff mentioned throwing in another check for if a registration has a DOI/ARK. If we need those let me know and I'll just modify the placeholder class property to check accordingly.

**Comment by sloria** [comment]
@JeffSpies What needs to be added for registrations with DOI/ARK?

**Comment by HarryRybacki** [comment]
@sloria modal 'irreversible' warning for making registrations registered post cutoff date is now in.

**Comment by HarryRybacki** [comment]
@sloria @JeffSpies anything else I need to refactor?

File: website/static/js/nodeControl.js [hunk 2]
```diff
@@ -35,7 +35,15 @@ var MESSAGES = {
 
     makeComponentPrivateWarning: 'Making a component private will prevent users from viewing it on this site, ' +
                         'but will have no impact on external sites, including Google\'s cache. ' +
-                        'Would you like to continue?'
+                        'Would you like to continue?',
+
+    // TODO(hrybacki): Remove once the hotifx is taken out
+    makePostCutoffRegistrationPublicWarning: 'Once a project is made public, there is no way to guarantee that ' +
```
Comment (sloria) [inline website/static/js/nodeControl.js]: ping @bnosek @JeffSpies for language review.

**Comment by sloria** [comment]
@HarryRybacki Can you also add the script to email affected users?

You can just use placeholder text for the email text for now. Follow up with @GaryKriebel for the final message text.

See website/mails.py for the email sending API.

**Comment by sloria** [comment]
Once the email script and language review is finished, this is good to merge.

**Comment by HarryRybacki** [comment]
What would the script do? This should be a one-time only email, right?

File: website/static/js/nodeControl.js (hunk 2 above)
Comment (JeffSpies) [inline website/static/js/nodeControl.js]: What lines are you asking to be reviewed? makePostCutoffRegistrationPublicWarning?

Comment (sloria) [inline website/static/js/nodeControl.js]: @JeffSpies Yes

**Comment by sloria** [comment]
@HarryRybacki The script should send an email to users who currently have a public registration, notifying that they will not be able to make their registrations private after the cutoff date. Correct me if I'm wrong here, @JeffSpies

**Comment by HarryRybacki** [comment]
Merged develop, not master into this on accident. Fixing.

File: website/static/js/nodeControl.js (hunk 2 above)
Comment (bnosek) [inline website/static/js/nodeControl.js]: looks like this was closed out, but confirming that language looks fine.

**Pull request closed without merging**